src/context/LiquidityWalletPageContext.tsx

```diff
@@ -212,45 +279,95 @@ export const LiquidityWalletPageContextProvider = ({
     if (
       parseFloat(amountAdd) >
       parseFloat(
-        wallets.find((wallet: LiquidityWallet) => id === wallet.id)?.tokens?.[
+        wallets.find((wallet: LiquidityWallet) => id === wallet._id)?.tokens?.[
           token
         ] || '0'
       )
     ) {
       setErrorMessage({
         type: 'amountAdd',
         text: `You can withdraw maximum ${
-          wallets.find((wallet: LiquidityWallet) => id === wallet.id)?.tokens?.[
-            token
-          ] || '0'
+          wallets.find((wallet: LiquidityWallet) => id === wallet._id)
+            ?.tokens?.[token] || '0'
         } tokens`,
       });
       return;
     }
+    if (!liquidityWalletAbi) {
+      setErrorMessage({
+        type: 'tx',
+        text: 'Contract ABI not found.',
+      });
+
+      return;
+    }
+
     setLoading(true);
+
+    const signer = provider.getSigner();
+
+    const wallet = await getWallet(id);
+
+    const _grtLiquidityWallet = new ethers.Contract(
+      wallet.walletAddress,
+      liquidityWalletAbi,
+      signer
+    );
+
+    const grtLiquidityWallet = _grtLiquidityWallet.connect(signer);
+
+    const txTransfer = await grtLiquidityWallet
```

Comment:
@tcoratger When I try to call the `withdrawNative` function i get the follow error: `execution reverted: Ownable: caller is not the owner"`

I checked and the Liquidity Wallet deploy is from this signer and the wallet address is correct as well.

Can you double check, please ?

Reply:
@juniormp  Please, provide some details for Thomas to be able to check it faster, in this case a link to transaction for the wallet creation, anything else related, specific parts of code, etc.

Reply:
@milukove @juniormp I think this fix `newContract.transferOwnership(msg.sender);` just added in the `deployLiquidityContract` function should do the job. I just deployed again :)

Reply:
FYI @juniormp @milukove just added a test here https://github.com/grindery-io/Depay-Reality/tree/47229da3da4515745 to avoid this error happening again